hnn_core/gui/gui.py

```diff
@@ -1118,9 +1130,41 @@ def load_drive_and_connectivity(params, log_out, drives_out,
                       layout)
 
 
-def on_upload_change(change, params, tstop, dt, log_out, drive_boxes,
-                     drive_widgets, drives_out, connectivity_out,
-                     connectivity_textfields, layout, load_type):
+def on_upload_data_change(change, data, viz_manager):
+    if len(change['owner'].value) == 0:
+        logger.info("Empty change")
+        return
+
+    key = list(change['new'].keys())[0]
+
+    data_fname = change['new'][key]['metadata']['name'].rstrip('.txt')
+    if data_fname in data['simulation_data'].keys():
+        logger.warn(f"Overwrite the existing data {data_fname}")
```

Comment:
This feels a bit dangerous ... would it be possible to allow users to name their data through a text box? Or not overwrite ... I can see how that would go wrong

Reply:
Renaming can be complicated, especially when the name is already used by other components like the visualization tab. I would recommend creating something like `{name} 1`, `{name} 2`, etc.

Reply:
How about preventing the user from uploading the same filename? If they try to do that, you can show a message (in the console) that they need to either rename the file or delete the already uploaded data.